apps/test-app/src/app/linked-query-param/linked-query-param-basic.component.ts

```diff
@@ -0,0 +1,72 @@
+import { JsonPipe } from '@angular/common';
+import { Component, effect, inject, untracked } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { Title } from '@angular/platform-browser';
+import { linkedQueryParam } from 'ngxtension/linked-query-param';
+
+@Component({
+	standalone: true,
+	template: `
+		<div>
+			<pre>SearchQuery: {{ searchQuery() | json }}</pre>
+
+			<div>
+				<label>
+					Different inputs same signal:
+					<input [(ngModel)]="searchQuery" name="a" placeholder="searchQuery" />
+					<input [(ngModel)]="searchQuery" name="b" placeholder="searchQuery" />
+				</label>
+			</div>
+
+			<!--			<div>-->
+			<!--				<label>-->
+			<!--					Different signals same query param:-->
+			<!--					<input-->
+			<!--						[(ngModel)]="differentSignalWithSearchQuery"-->
+			<!--						name="c"-->
+			<!--						placeholder="differentSignalWithSearchQuery"-->
+			<!--					/>-->
+			<!--				</label>-->
+			<!--			</div>-->
+
+			<button type="button" (click)="searchQuery.set('cool')">cool</button>
+			<button type="button" (click)="searchQuery.set('great')">great</button>
+			<button type="button" (click)="searchQuery.set(null)">Reset</button>
+		</div>
+	`,
+	imports: [FormsModule, JsonPipe],
+	styles: `
+		div {
+			padding: 20px;
+		}
+	`,
+})
+export default class LinkedQueryParamStringCmp {
+	private titleService = inject(Title);
+
+	searchQuery = linkedQueryParam('searchQuery', {
+		// NOTE in docs that serialize should come after parse in order for types to work correctly
+		// parse: (x) => x ? parseInt(x, 10) : null,
+		// serialize: (x) => x,
+	});
+
+	// differentSignalWithSearchQuery = linkedQueryParam('searchQuery', {
+	// 	defaultValue: 'default',
+	// });
+
+	constructor() {
+		effect(() => {
+			console.log('searchQuery Type: ', typeof this.searchQuery());
+			console.log('searchQuery Value: ', this.searchQuery());
+
+			untracked(() => {
+				const searchQuery = this.searchQuery();
+				if (searchQuery) {
+					this.titleService.setTitle(searchQuery);
+				} else {
+					this.titleService.setTitle('No search query!');
+				}
+			});
+		});
```

Comment:
nit: this seems a bit confusing to me. The `effect` seems to rely on `searchQuery` to update the `title` but the actual `searchQuery` value is in `untracked`? I am aware that the track of `searchQuery()` happens in the `console.log`

Reply:
forgot about this one. refactored